test/apply/apply.test.ts

```diff
@@ -0,0 +1,353 @@
+  ],
+  [
+    'set_node',
+    [
+      createNode('paragraph', 'first', { test: '1234' }),
+      createNode('paragraph', 'second'),
+    ],
+    [
+      {
+        path: [0],
+        type: 'set_node',
+        properties: {},
+        newProperties: {
+          test: '4567',
+        },
+      },
+    ],
+    [
+      createNode('paragraph', 'first', { test: '4567' }),
+      createNode('paragraph', 'second'),
+    ],
+  ],
+  [
+    'split_node',
+    [createNode('paragraph', 'Hello collaborator!')],
+    [
+      {
+        path: [0, 0],
+        position: 6,
+        properties: {},
+
+        type: 'split_node',
+      },
+      {
+        path: [0],
+        position: 1,
+        properties: {
+          type: 'paragraph',
+        },
+        target: 6,
+        type: 'split_node',
+      },
+    ],
+    [
+      createNode('paragraph', 'Hello '),
+      createNode('paragraph', 'collaborator!'),
+    ],
+  ],
+  [
+    'add_mark',
+    [createNode('paragraph', 'hotel uniform golf oscar')],
+    [
+      {
+        path: [0, 0],
+        position: 6,
+        properties: {},
+
+        type: 'split_node',
+      },
+      {
+        path: [0, 0],
+        properties: {},
+        newProperties: {
+          em: true,
+        },
+        type: 'set_node',
+      },
+      {
+        path: [0, 1],
+        position: 7,
+        properties: {},
+
+        type: 'split_node',
+      },
+      {
+        path: [0, 1],
+        properties: {},
+        newProperties: {
+          em: true,
+          strong: true,
+        },
+        type: 'set_node',
+      },
+      {
+        path: [0, 2],
+        position: 5,
+        properties: {},
+
+        type: 'split_node',
+      },
+      {
+        path: [0, 2],
+        properties: {},
+        newProperties: {
+          strong: true,
+        },
+        type: 'set_node',
+      },
+    ],
+    [
+      {
+        type: 'paragraph',
+        children: [
+          { text: 'hotel ', em: true },
+          { text: 'uniform', em: true, strong: true },
+          { text: ' golf', strong: true },
+          { text: ' oscar' },
+        ],
+      },
+    ],
+  ],
+  [
+    'remove_mark',
+    [
+      {
+        type: 'paragraph',
+        children: [
+          { text: 'hotel ', em: true },
+          { text: 'uniform', em: true, strong: true },
+          { text: ' golf', strong: true },
+          { text: ' oscar' },
+        ],
+      },
+    ],
+    [
+      {
+        path: [0, 0],
+        properties: { em: true },
+        newProperties: {},
+        type: 'set_node',
+      },
+      {
+        path: [0, 1],
+        properties: { em: true, strong: true },
+        newProperties: {},
+        type: 'set_node',
+      },
+      {
+        path: [0, 2],
+        properties: { strong: true },
+        newProperties: {},
+        type: 'set_node',
+      },
+    ],
+    [createNode('paragraph', 'hotel uniform golf oscar')],
+  ],
+  [
+    // TODO: check the new way to add text with marks
```

Comment:
Anything I need to test in the UI?

Reply:
The only thing would be confirm if the test steps is robust enough with the real usage situation.